compose/snippets/src/main/java/com/example/compose/snippets/pictureInPicture/PictureInPictureSnippets.kt

```diff
@@ -0,0 +1,384 @@
+                builder.setSourceRectHint(sourceRect)
+                builder.setAspectRatio(
+                    Rational(sourceRect.width(), sourceRect.height())
+                )
+            }
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                builder.setAutoEnterEnabled(shouldEnterPipMode)
+            }
+            context.findActivity().setPictureInPictureParams(builder.build())
+        }
+    }
+    // [END android_compose_pip_set_aspect_ratio]
+}
+
+// [START android_compose_pip_build_remote_action]
+@RequiresApi(Build.VERSION_CODES.O)
+private fun buildRemoteAction(
+    @DrawableRes iconResId: Int,
+    @StringRes titleResId: Int,
+    requestCode: Int,
+    controlType: Int,
+    context: Context
+): RemoteAction {
+    return RemoteAction(
+        Icon.createWithResource(context, iconResId),
+        context.getString(titleResId),
+        context.getString(titleResId),
+        PendingIntent.getBroadcast(
+            context,
+            requestCode,
+            Intent(ACTION_BROADCAST_CONTROL)
+                .putExtra(EXTRA_CONTROL_TYPE, controlType),
+            PendingIntent.FLAG_IMMUTABLE
+        )
+    )
+}
+// [END android_compose_pip_build_remote_action]
+
+// [START android_compose_pip_broadcast_receiver]
+@RequiresApi(Build.VERSION_CODES.O)
+@Composable
+fun BroadcastReceiver(player: Player?) {
+    if (isInPipMode() && player != null) {
+        val context = LocalContext.current
+
+        DisposableEffect(key1 = player) {
+            val broadcastReceiver: BroadcastReceiver = object : BroadcastReceiver() {
+                override fun onReceive(context: Context?, intent: Intent?) {
+                    if ((intent == null) || (intent.action != ACTION_BROADCAST_CONTROL)) {
+                        return
+                    }
+
+                    when (intent.getIntExtra(EXTRA_CONTROL_TYPE, 0)) {
+                        EXTRA_CONTROL_PAUSE -> player.pause()
+                        EXTRA_CONTROL_PLAY -> player.play()
+                    }
+                }
+            }
+            ContextCompat.registerReceiver(
+                context,
+                broadcastReceiver,
+                IntentFilter(ACTION_BROADCAST_CONTROL),
+                ContextCompat.RECEIVER_NOT_EXPORTED
+            )
+            onDispose {
+                context.unregisterReceiver(broadcastReceiver)
+            }
+        }
+    }
+}
+// [END android_compose_pip_broadcast_receiver]
+
+@RequiresApi(Build.VERSION_CODES.O)
+fun listOfRemoteActions(isPlaying: Boolean, context: Context): List<RemoteAction> {
+    return listOf()
+}
+
+@Composable
+fun VideoPlayer4(
+    shouldEnterPipMode: Boolean,
+    modifier: Modifier = Modifier,
+
+) {
+    // [START android_compose_pip_add_remote_actions]
+    val context = LocalContext.current
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
What if the version is less than `O`? Should there be an `else` case?

Reply:
If >= `O` is required, maybe add a `RequiresApi`?

Reply:
ah - RequiresAPI may be a better option here